table/block_based/block_based_table_builder.cc

```diff
@@ -788,6 +1009,60 @@ void BlockBasedTableBuilder::WriteRawBlock(const Slice& block_contents,
   }
 }
 
+void BlockBasedTableBuilder::BGWorkWriteRawBlock() {
+  Rep* r = rep_;
+  ParallelCompressionRep::BlockRepSlot* slot;
+  ParallelCompressionRep::BlockRep* block_rep;
+  while (r->pc_rep->write_queue.pop(slot)) {
+    slot->Take(block_rep);
+
+    for (size_t i = 0; i < block_rep->keys->Size(); i++) {
+      auto& key = (*block_rep->keys)[i];
+      if (r->filter_builder != nullptr) {
+        size_t ts_sz =
+            r->internal_comparator.user_comparator()->timestamp_size();
+        r->filter_builder->Add(ExtractUserKeyAndStripTimestamp(key, ts_sz));
+      }
+      r->index_builder->OnKeyAdded(key);
+    }
+
+    WriteRawBlock(block_rep->contents, block_rep->compression_type,
+                  &r->pending_handle, true /* is_data_block*/);
+
+    // Update raw_bytes_compressed after WriteRawBlock() to ensure inflight
+    // bytes are written
+    r->pc_rep->curr_compression_ratio =
+        (r->pc_rep->curr_compression_ratio * r->pc_rep->raw_bytes_compressed +
+         block_rep->compressed_data->size()) /
+        double(r->pc_rep->raw_bytes_compressed + block_rep->data->size());
```

Comment:
We follow Google C++ Style, which bans C style casting: https://google.github.io/styleguide/cppguide.html#Casting try to use a C++ style casting instead.

Reply:
Fixed.